src/web/controllers/admin/player_admin_controller.py

```diff
@@ -471,72 +454,72 @@ def sort_key(player: Player): # pylint: disable=function-redefined
+            for player in web_context.admin_event.players_by_id.values()
+            if (
+                player.id is not None
+                and player.ref_id > 1
+                and len(filter_genders) in [0, 3]
+                or player.gender.value in filter_genders
+            )
+            and (
+                len(filter_categories) in [0, len(players_categories)]
+                or player.category in filter_categories
+            )
+            and (
+                len(filter_check_ins) in [0, 3]
+                or (
+                    player.can_check_in_out
+                    and player.check_in in filter_check_ins
+                )
+                or (
+                    not player.can_check_in_out and None in filter_check_ins
+                )
+            )
+            and (
+                len(filter_tournaments)
+                in [0, len(web_context.admin_event.tournaments_by_id)]
+                or player.tournament_id in filter_tournaments
+            )
+            and (
+                len(filter_federations) in [0, len(players_federations)]
+                or player.federation in filter_federations
+            )
+            and (
+                len(filter_clubs) in [0, len(players_clubs)]
+                or player.club in filter_clubs
+            )
+            and all(
+                {
+                    filter_name_part
+                    in unicode_normalize(
+                        f'{player.last_name} {player.first_name}'.lower()
                     )
-                    and all(
-                        {
-                            filter_origin_part
-                            in unicode_normalize(
-                                f'{player.federation} {player.club}'.lower()
-                            )
-                            for filter_origin_part in filter_origin_parts
-                        }
+                    for filter_name_part in filter_name_parts
+                }
+            )
+            and all(
+                {
+                    filter_origin_part
+                    in unicode_normalize(
+                        f'{player.federation} {player.club}'.lower()
                     )
-                    and all(plugin_manager.hook.filter_player(
-                        web_context=web_context,
-                        template_context=template_context,
-                        player=player
-                    ))
-                ],
-                key=sort_key,
+                    for filter_origin_part in filter_origin_parts
+                }
             )
-        }
+            and all(plugin_manager.hook.filter_player(
+                web_context=web_context,
+                template_context=template_context,
+                player=player
+            ))
+        ]
         
+        players: dict[int, Player] = {}
+        for player in sorted(filtered_players, key=get_sort_key):
+            if player.id is not None:
```

Comment:
hasn't the `filtered_players` already filtered out the players with no id?

Reply:
The type system can't cope here